R/addins.R

```diff
@@ -105,6 +112,31 @@ style_selection <- function() {
   }
 }
 
+style_selection_v2 <- function() {
+  communicate_addins_style_transformers()
+  id <- rstudioapi::documentId()
+  selection <- rstudioapi::selectionGet(id = id)
+  text <- selection$value
+  if (!nzchar(text)) abort("No code selected")
+  out <- style_text(text, transformers = get_addins_style_transformer())
+  rstudioapi::selectionSet(value = out, id = id)
```

Comment:
What we did above with `if (context$selection[[1]]$range$end[2] == 1) "")` guarantees that when your selection is 
![Screenshot 2020-09-18 at 23 44 40](https://user-images.githubusercontent.com/10477073/93647534-ee205480-fa08-11ea-9d94-e5a60ad46237.png)

There is exactly one line break after the selected area (and no blank line), and if it is 
![Screenshot 2020-09-18 at 23 45 06](https://user-images.githubusercontent.com/10477073/93647558-fc6e7080-fa08-11ea-8746-e3447a48f017.png)
there is also a line break after the selection and no blank line. Does your implementation guarantee this as well?

Reply:
It doesn't; the new API doesn't provide any mechanism for expanding on modifying the user's current selection currently. We should probably discuss this on https://github.com/rstudio/rstudio/issues/7465.